controllers/datafoundationclient.go

```diff
@@ -248,3 +275,46 @@ func (r *dataFoundationClientReconciler) reconcileCSIKMSConnectionDetailsConfigM
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := map[string]bool{}
+	rbdProvisionerName := r.offering.Namespace + ocsRBDProvisionerSuffix
+	fsProvisionerName := r.offering.Namespace + ocsCephFsProvisionerSuffix
+	for i := range storageClassList.Items {
+		storageClass := storageClassList.Items[i]
+		if storageClass.Provisioner == fsProvisionerName || storageClass.Provisioner == rbdProvisionerName {
+			ocsStorageClass[storageClass.Name] = true
+		}
+	}
+	pvList := &corev1.PersistentVolumeList{}
+	if err := r.UnrestrictedClient.List(r.ctx, pvList); err != nil {
+		return false, fmt.Errorf("failed to list persistent volumes: %v", err)
+	}
+	for i := range pvList.Items {
+		scName := pvList.Items[i].Spec.StorageClassName
+		if ocsStorageClass[scName] {
+			return true, nil
+		}
+	}
+	return false, nil
```

Comment:
Shouldn't we list all the PVs and related storageClass that are preventing the uninstallation of the offering?
Because now customers can create multiple storageClassClaims so it will be difficult to track which PV is blocking the uninstallation of offering.

Reply:
@bindrad So the ask is to list all the PVS blocking unistallation and log them?
Currently the code will map all the storageclasses provisioned using cephfs.csi.ceph.com or rbd.csi.ceph.com
later use this map to find if any pvs are using the said storageclasses.

Reply:
Yes, I think we should list all the PVs that are using ocsStorageClass as that is useful information to unblock uninstallation.

Reply:
Why should we log them? Who will read that log?

Reply:
Based on past experiences, what if SRE's don't have permission to list all the PVs in the cluster?
If we are logging the info, SRE can send the list to the customer asking to delete the PVs.

> put a potentially very big list of names in our log every few seconds

We will log the list only when uninstallation is triggered and if uninstallation is blocked because of a specific set of PVs in the cluster then I think that is a useful piece of information.

Reply:
Sounds good. I will update the implementation to log the uninstallation blocking PVS

Reply:
After a discussion with Ohad regarding this, I think logging number of PVs should be good enough as there can be 100s of PVs so we will be logging a huge list. Instead SREs can view the PVs on cloud monitoring if required.

Reply:
updatded with pvCount